engine/transaction/draft/create_draft.go

```diff
@@ -0,0 +1,35 @@
+package draft
+
+import (
+	"context"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+)
+
+// Create creates a new draft transaction based on specification.
+func Create(ctx context.Context, spec *TransactionSpec) (*Transaction, error) {
+	tx := &sdk.Transaction{}
+	if spec == nil {
+		return nil, txerrors.ErrDraftSpecificationRequired
+	}
+	outputs, annotations, err := spec.outputs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tx.Outputs = outputs
+
+	beef, err := tx.BEEFHex()
+	if err != nil {
```

Comment:
Shouldn't we validate the beef here? At least if it's not a ""? 
I would not fully rely on sdk checks as we have no control of the code there.

Reply:
I cannot agree with that, we must start believing them and eventually either request changes/fixes or make CR with it.
We are playing to the same goal, and we want sdk to be fully reliable.

Reply:
Okay, I won't argue :D